pkg/hooks/http/http.go

```diff
@@ -76,8 +100,15 @@ func (h HttpHook) InvokeHook(hookReq hooks.HookRequest) (hookRes hooks.HookRespo
 		return hookRes, fmt.Errorf("unexpected response code from hook endpoint (%d): %s", httpRes.StatusCode, string(httpBody))
 	}
 
-	if err = json.Unmarshal(httpBody, &hookRes); err != nil {
-		return hookRes, fmt.Errorf("failed to parse hook response: %w", err)
+	decoder := json.NewDecoder(bytes.NewReader(httpBody))
```

Comment:
Why did you decide to change this to `json.NewDecoder` versus keeping `json.Unmarshal`? Creating a new reader seems unnecessary as the response has already been read into a buffer.